ml_metadata/tools/mlmd_bench/fill_context_edges_workload.cc

```diff
@@ -0,0 +1,245 @@
+      << "Unexpected Types";
+  int64 i = 0;
+  std::default_random_engine gen;
+  while (i < num_edges) {
+    const int64 non_context_node_index = non_context_node_index_dist(gen);
+    const int64 context_node_index = context_node_index_dist(gen);
+    int64 non_context_node_id;
+    if (std::is_same<T, Attribution>::value) {
+      non_context_node_id =
+          absl::get<Artifact>(
+              existing_non_context_nodes[non_context_node_index])
+              .id();
+    } else if (std::is_same<T, Association>::value) {
+      non_context_node_id =
+          absl::get<Execution>(
+              existing_non_context_nodes[non_context_node_index])
+              .id();
+    }
+    const int64 context_node_id =
+        absl::get<Context>(existing_context_nodes[context_node_index]).id();
+
+    bool already_existed_in_current_setup =
+        CheckDuplicateContextEdgeInCurrentSetUp(
+            non_context_node_id, context_node_id, unique_checker);
+    tensorflow::Status status = CheckDuplicateContextEdgeInDb<T>(
+        non_context_node_id, context_node_id, store);
+    if (!status.ok() && status.code() != tensorflow::error::ALREADY_EXISTS) {
+      return status;
+    }
+    // Rejection sampling.
+    if (already_existed_in_current_setup || !status.ok()) {
+      continue;
+    }
+    SetCurrentContextEdge<T>(non_context_node_id, context_node_id, request);
+    // Increases `curr_bytes` with the size of `non_context_node_id` and 
+    // `context_node_id`.
+    curr_bytes += 8 * 2;
+    i++;
+  }
+  return tensorflow::Status::OK();
+}
+
+}  // namespace
+
+FillContextEdges::FillContextEdges(
+    const FillContextEdgesConfig& fill_context_edges_config,
+    int64 num_operations)
+    : fill_context_edges_config_(fill_context_edges_config),
+      num_operations_(num_operations),
+      name_(absl::StrCat("FILL_",
+                         fill_context_edges_config_.Specification_Name(
+                             fill_context_edges_config_.specification()))) {}
+
+tensorflow::Status FillContextEdges::SetUpImpl(MetadataStore* store) {
+  LOG(INFO) << "Setting up ...";
+
+  int64 curr_bytes = 0;
+
+  std::vector<Node> existing_non_context_nodes;
+  std::vector<Node> existing_context_nodes;
+  TF_RETURN_IF_ERROR(GetExistingNodes(fill_context_edges_config_, *store,
+                                      existing_non_context_nodes,
+                                      existing_context_nodes));
+
+  const double non_context_alpha =
+      fill_context_edges_config_.non_context_node_popularity()
+          .dirichlet_alpha();
+  const double context_alpha =
+      fill_context_edges_config_.context_node_popularity().dirichlet_alpha();
+  // Generates categorical distribution with a Dirichlet distribution with
+  // a concentrate parameter.
+  std::discrete_distribution<int64> non_context_node_index_dist =
+      GenerateCategoricalDistributionWithDirichletPrior(
+          existing_non_context_nodes.size(), non_context_alpha);
```

Comment:
pass a config instead of alpha

Reply:
Yes, since GenerateCategoricalDistributionWithDirichletPrior() will also be used in FillEvents workload later, I think it may be better to pass in a alpha instead of a concrete type of config. Wdyt?